### PR TITLE
Replace Preview Download controller with webapi endpoint

### PR DESCRIPTION
**Pull request by engcom-Golf**
### Description (*)
Replaced Preview Download controller with webapi endpoint

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#283: Replace Preview Download controller with webapi endpoint

### Manual testing scenarios (*)

1. Go to Category -> Content-> Insert Image -> Adobe Stock Image -> Try To save Preview

### CODE REVIEW

File: AdobeStockImage/Model/SaveImagePreview.php [hunk 1]
```diff
@@ -113,11 +110,13 @@ public function execute(int $adobeId, string $destinationPath): void
             $path = $this->storage->save($asset->getPreviewUrl(), $destinationPath);
             $asset->setPath($path);
             $this->saveAsset($asset);
+            return $asset->getThumbnailUrl();
```
Comment (sivaschenko) [inline AdobeStockImage/Model/SaveImagePreview.php]: The returned URL should be a Magento URL, i.e. `local.magento.com/media/file.png`. It should be retrieved based on asset path

**Review by sivaschenko — CHANGES_REQUESTED** [review]
@engcom-Golf thanks for the PR, please see my comment

File: AdobeStockImage/Model/SaveImagePreview.php (hunk 1 above)
Comment (Nazar65) [reply AdobeStockImage/Model/SaveImagePreview.php]: Ok

Comment (sivaschenko) [reply AdobeStockImage/Model/SaveImagePreview.php]: @Nazar65 after changing the frontend approach in https://github.com/magento/adobe-stock-integration/pull/302 the service no longer has to return the url.

It should `return void`

Comment (Nazar65) [reply AdobeStockImage/Model/SaveImagePreview.php]: Ok i will update.

**Comment by sivaschenko** [comment]
Blocked by https://github.com/magento/magento2/issues/9138

File: AdobeStockImage/etc/webapi.xml [hunk 2]
```diff
@@ -13,4 +13,10 @@
             <resource ref="Magento_AdobeStockAsset::actions_view"/>
         </resources>
     </route>
+    <route url="/V1/adobestock/saveImagePreview" method="POST">
+        <service class="Magento\AdobeStockImageApi\Api\SaveImagePreviewInterface" method="execute"/>
+        <resources>
+            <resource ref="anonymous"/>
```
Comment (sivaschenko) [inline AdobeStockImage/etc/webapi.xml:19]: This WebAPI endpoint should be protected by an appropriate ACL

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @engcom-Golf ! Please see my review comments

**Comment by sivaschenko** [comment]
We are unable to implement this change without a proper webapi ACL and admin session authentication support from Magento framework.

Closing this PR considering the above.

**Pull request closed without merging**